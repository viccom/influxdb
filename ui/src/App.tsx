import React, {SFC, ReactChildren} from 'react'

import RightClickLayer from 'src/clockface/components/right_click_menu/RightClickLayer'
import Nav from 'src/pageLayout'
import Notifications from 'src/shared/components/notifications/Notifications'
import LegendPortal from 'src/shared/components/LegendPortal'

interface Props {
  children: ReactChildren
}

const App: SFC<Props> = ({children}) => (
  <div className="chronograf-root">
    <Notifications />
    <RightClickLayer />
    <Nav />
    <LegendPortal />
    {children}
  </div>
)

export default App
