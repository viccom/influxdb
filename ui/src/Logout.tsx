// Libraries
import {PureComponent} from 'react'
import {withRouter, WithRouterProps} from 'react-router'

// APIs
import {logout} from 'src/me/apis'

// Components
import {ErrorHandling} from 'src/shared/decorators/errors'

type Props = WithRouterProps

@ErrorHandling
export class Logout extends PureComponent<Props> {
  public componentDidMount() {
    this.handleSignOut()
  }

  public render() {
    return null
  }

  private handleSignOut = async () => {
    await logout()
    this.props.router.push('/')
  }
}

export default withRouter<Props>(Logout)
