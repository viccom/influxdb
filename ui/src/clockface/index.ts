// Import Components
import AutoInput from './components/auto_input/AutoInput'
import Button from './components/Button'
import ConfirmationButton from './components/confirmation_button/ConfirmationButton'
import Dropdown, {DropdownMode} from './components/dropdowns/Dropdown'
import MultiSelectDropdown from './components/dropdowns/MultiSelectDropdown'
import Form from './components/form_layout/Form'
import Input, {InputType, AutoComplete} from './components/inputs/Input'
import MultipleInput, {
  MultiInputType,
} from './components/inputs/multipleInput/MultipleInput'
import OverlayTechnology from './components/overlays/OverlayTechnology'
import OverlayContainer from './components/overlays/OverlayContainer'
import OverlayHeading from './components/overlays/OverlayHeading'
import OverlayBody from './components/overlays/OverlayBody'
import OverlayFooter from './components/overlays/OverlayFooter'
import Panel from './components/panel/Panel'
import Radio from './components/radio_buttons/RadioButtons'
import SlideToggle from './components/slide_toggle/SlideToggle'
import WizardFullScreen from './components/wizard/WizardFullScreen'
import WizardOverlay from './components/wizard/WizardOverlay'
import WizardProgressHeader from './components/wizard/WizardProgressHeader'
import ProgressBar from './components/wizard/ProgressBar'
import ComponentSpacer from './components/component_spacer/ComponentSpacer'
import EmptyState from './components/empty_state/EmptyState'
import Spinner from './components/spinners/Spinner'
import SparkleSpinner from './components/spinners/SparkleSpinner'
import IndexList from './components/index_views/IndexList'
import Context from './components/context_menu/Context'
import FormElement from 'src/clockface/components/form_layout/FormElement'
import DraggableResizer from 'src/clockface/components/draggable_resizer/DraggableResizer'
import Label, {LabelType} from 'src/clockface/components/label/Label'
import LabelSelector from 'src/clockface/components/label/LabelSelector'
import GridSizer from 'src/clockface/components/grid_sizer/GridSizer'
import Select from 'src/clockface/components/Select'
import Grid from 'src/clockface/components/grid_layout/Grid'

// Import Types
import {
  ComponentColor,
  ComponentSize,
  ComponentStatus,
  DropdownMenuColors,
  DropdownChild,
  ButtonShape,
  Greys,
  IconFont,
  Columns,
  Alignment,
  Stack,
  ButtonType,
  Sort,
} from './types'

// Fire de lazer
export {
  Alignment,
  AutoComplete,
  AutoInput,
  Button,
  ButtonType,
  ButtonShape,
  Columns,
  ComponentColor,
  ComponentSize,
  ComponentSpacer,
  ComponentStatus,
  ConfirmationButton,
  Context,
  DraggableResizer,
  Dropdown,
  DropdownChild,
  DropdownMenuColors,
  DropdownMode,
  EmptyState,
  Form,
  FormElement,
  Greys,
  Grid,
  GridSizer,
  IconFont,
  IndexList,
  Input,
  InputType,
  Label,
  LabelSelector,
  LabelType,
  MultiSelectDropdown,
  MultiInputType,
  MultipleInput,
  OverlayBody,
  OverlayContainer,
  OverlayFooter,
  OverlayHeading,
  OverlayTechnology,
  Panel,
  ProgressBar,
  Radio,
  Select,
  SlideToggle,
  SparkleSpinner,
  Sort,
  Spinner,
  Stack,
  WizardFullScreen,
  WizardProgressHeader,
  WizardOverlay,
}
