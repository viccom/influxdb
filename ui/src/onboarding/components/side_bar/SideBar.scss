/*
  Side Bar Styles
  ------------------------------------------------------------------------------
*/

@import 'src/style/modules';

$sidebar-width-sm: 140px;
$sidebar-width-md: 240px;

.side-bar {
  @include no-user-select();
  overflow: hidden;
  transform: translate3d(0,0,0);
  flex: 1 0 0;
  transition: flex 0.4s ease;

  &.show {
    flex: 1 0 $sidebar-width-sm;
  }
}

.side-bar--container {
  box-sizing: border-box;
  opacity: 0;
  transition: opacity 0.5s ease;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  padding: $ix-marg-e 0 34px 0;
}

.side-bar.show .side-bar--container {
  opacity: 1;
}

.side-bar--title {
  margin: 0 $ix-marg-c $ix-marg-c $ix-marg-c;
  color: $g13-mist;
  font-size: $ix-text-base-1;
}

.side-bar--contents {
  display: flex;
  flex: 1 0 0;
  flex-direction: column;
  justify-content: space-between;
}

.side-bar--tabs {
  display: flex;
  flex-direction: column;
  align-items: stretch;
}

.side-bar--tab {
  height: 44px;
  display: flex;
  align-items: center;
  border-radius: $radius 0 0 $radius;
  font-size: 15px;
  font-weight: 600;
  margin-left: $ix-marg-c;
  color: $g10-wolf;
  background-color: transparent;
  white-space: nowrap;
  transition: color 0.25s ease, background-color 0.25s ease;

  &:hover {
    color: $g15-platinum;
    cursor: pointer;
    background-color: rgba($g3-castle, 0.6);
  }

  &.active {
    color: $g15-platinum;
    background-color: $g3-castle;
  }
}

.side-bar--icon {
  width: 16px;
  color: $g7-graphite;
  display: inline-block;
  font-size: 16px;
  margin: 0 13px;
  transition: color 0.25s ease;

  .side-bar--tab:hover & {
    color: $g11-sidewalk;
  }

  .side-bar--tab.active & {
    color: $c-pool;
  }
}

// Tab Status Coloration
.side-bar--tab__success,
.side-bar--tab__success:hover {
  &,
  & .side-bar--icon {
    color: $c-rainforest;
  }
}

.side-bar--tab__pending,
.side-bar--tab__pending:hover {
  &,
  & .side-bar--icon {
    color: $c-pineapple;
  }
}

.side-bar--tab__error,
.side-bar--tab__error:hover {
  &,
  & .side-bar--icon {
    color: $c-dreamsicle;
  }
}

.side-bar--buttons {
  margin: 0 $ix-marg-c;
}

.side-bar--button {
  width: 100%;
}

@media screen and (min-width: $grid--breakpoint-md) {
  .side-bar--tab {
    margin-left: $ix-marg-d;
  }

  .side-bar--title {
    margin: 0 $ix-marg-d $ix-marg-c $ix-marg-d;
    font-size: $ix-text-base-2;
  }

  .side-bar--buttons {
    margin: 0 $ix-marg-d;
  }

  .side-bar {
    &.show {
      flex: 1 0 $sidebar-width-md;
    }
  }
}