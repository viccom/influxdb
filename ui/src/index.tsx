import 'babel-polyfill'

import React, {PureComponent} from 'react'
import {render} from 'react-dom'
import {Provider} from 'react-redux'
import {Router, Route, useRouterHistory, IndexRoute} from 'react-router'
import {createHistory, History} from 'history'

import configureStore from 'src/store/configureStore'
import {loadLocalStorage} from 'src/localStorage'

import {getRootNode} from 'src/utils/nodes'
import {getBasepath} from 'src/utils/basepath'

// Components
import App from 'src/App'
import GetSources from 'src/shared/containers/GetSources'
import SetActiveSource from 'src/shared/containers/SetActiveSource'
import GetOrganizations from 'src/shared/containers/GetOrganizations'
import Setup from 'src/Setup'
import Signin from 'src/Signin'
import SigninPage from 'src/onboarding/containers/SigninPage'
import Logout from 'src/Logout'
import TaskPage from 'src/tasks/containers/TaskPage'
import TasksPage from 'src/tasks/containers/TasksPage'
import OrganizationsIndex from 'src/organizations/containers/OrganizationsIndex'
import OrganizationView from 'src/organizations/containers/OrganizationView'
import TaskEditPage from 'src/tasks/containers/TaskEditPage'
import DashboardPage from 'src/dashboards/components/DashboardPage'
import DashboardsIndex from 'src/dashboards/components/dashboard_index/DashboardsIndex'
import DataExplorerPage from 'src/dataExplorer/components/DataExplorerPage'
import {MePage, Account} from 'src/me'
import NotFound from 'src/shared/components/NotFound'
import GetLinks from 'src/shared/containers/GetLinks'
import GetMe from 'src/shared/containers/GetMe'
import SourcesPage from 'src/sources/components/SourcesPage'
import ConfigurationPage from 'src/configuration/components/ConfigurationPage'

import OnboardingWizardPage from 'src/onboarding/containers/OnboardingWizardPage'

// Actions
import {disablePresentationMode} from 'src/shared/actions/app'

// Styles
import 'src/style/chronograf.scss'

const rootNode = getRootNode()
const basepath = getBasepath()

declare global {
  interface Window {
    basepath: string
  }
}

// Older method used for pre-IE 11 compatibility
window.basepath = basepath

const history: History = useRouterHistory(createHistory)({
  basename: basepath, // this is written in when available by the URL prefixer middleware
})

const store = configureStore(loadLocalStorage(), history)
const {dispatch} = store

history.listen(() => {
  dispatch(disablePresentationMode())
})

window.addEventListener('keyup', event => {
  const escapeKeyCode = 27
  // fallback for browsers that don't support event.key
  if (event.key === 'Escape' || event.keyCode === escapeKeyCode) {
    dispatch(disablePresentationMode())
  }
})

class Root extends PureComponent {
  public render() {
    return (
      <Provider store={store}>
        <Router history={history}>
          <Route component={GetLinks}>
            <Route component={Setup}>
              <Route path="/onboarding">
                <Route path=":stepID" component={OnboardingWizardPage} />
                <Route
                  path=":stepID/:substepID"
                  component={OnboardingWizardPage}
                />
              </Route>
              <Route path="/signin" component={SigninPage} />
              <Route path="/logout" component={Logout} />
              <Route path="/">
                <Route component={Signin}>
                  <Route component={GetMe}>
                    <Route component={GetOrganizations}>
                      <Route component={App}>
                        <Route component={GetSources}>
                          <Route component={SetActiveSource}>
                            <IndexRoute component={DashboardsIndex} />
                            <Route path="organizations">
                              <IndexRoute component={OrganizationsIndex} />
                              <Route
                                path=":orgID/:tab"
                                component={OrganizationView}
                              />
                            </Route>
                            <Route path="tasks">
                              <IndexRoute component={TasksPage} />
                              <Route path="new" component={TaskPage} />
                              <Route path=":id" component={TaskEditPage} />
                            </Route>
                            <Route
                              path="data-explorer"
                              component={DataExplorerPage}
                            />
                            <Route path="dashboards">
                              <IndexRoute component={DashboardsIndex} />
                              <Route
                                path=":dashboardID"
                                component={DashboardPage}
                              />
                            </Route>
                            <Route path="me" component={MePage} />
                            <Route path="account/:tab" component={Account} />
                            <Route path="sources" component={SourcesPage} />
                            <Route
                              path="configuration/:tab"
                              component={ConfigurationPage}
                            />
                          </Route>
                        </Route>
                      </Route>
                    </Route>
                  </Route>
                </Route>
              </Route>
            </Route>
          </Route>
          <Route path="*" component={NotFound} />
        </Router>
      </Provider>
    )
  }
}

if (rootNode) {
  render(<Root />, rootNode)
}
