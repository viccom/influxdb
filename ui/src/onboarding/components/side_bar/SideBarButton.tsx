// Libraries
import React, {Component} from 'react'

// Components
import {Button, ComponentColor, ComponentSize} from 'src/clockface'

// Types
import {IconFont} from 'src/clockface'

interface Props {
  text: string
  titleText: string
  color: ComponentColor
  icon?: IconFont
  onClick?: () => void
}

class SideBarButton extends Component<Props> {
  public render() {
    const {text, titleText, color, onClick, icon} = this.props

    return (
      <Button
        customClass="side-bar--button"
        text={text}
        titleText={titleText}
        onClick={onClick}
        size={ComponentSize.Small}
        color={color}
        icon={icon}
      />
    )
  }
}

export default SideBarButton
