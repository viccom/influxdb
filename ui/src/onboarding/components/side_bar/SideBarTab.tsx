// Libraries
import React, {Component} from 'react'
import classnames from 'classnames'

// Types
import {IconFont} from 'src/clockface'
import {SideBarTabStatus as TabStatus} from 'src/onboarding/components/side_bar/SideBar'

interface Props {
  label: string
  id: string
  active: boolean
  status: TabStatus
  onClick: (tabID: string) => void
}

class SideBarTab extends Component<Props> {
  public render() {
    const {label} = this.props

    return (
      <div className={this.className} onClick={this.handleClick}>
        {this.icon}
        {label}
      </div>
    )
  }

  private handleClick = (): void => {
    const {id, onClick} = this.props

    onClick(id)
  }

  private get className(): string {
    const {status, active} = this.props

    return classnames('side-bar--tab', {
      active,
      'side-bar--tab__success': status === TabStatus.Success,
      'side-bar--tab__error': status === TabStatus.Error,
      'side-bar--tab__pending': status === TabStatus.Pending,
    })
  }

  private get icon(): JSX.Element {
    const {status} = this.props
    let icon

    switch (status) {
      case TabStatus.Pending:
      case TabStatus.Success:
        icon = `side-bar--icon icon ${IconFont.Checkmark}`
        break
      case TabStatus.Error:
        icon = `side-bar--icon icon ${IconFont.Remove}`
        break
      case TabStatus.Default:
        icon = `side-bar--icon icon ${IconFont.CircleThick}`
        break
      default:
        icon = `side-bar--icon`
    }

    return <span className={icon} />
  }
}

export default SideBarTab
