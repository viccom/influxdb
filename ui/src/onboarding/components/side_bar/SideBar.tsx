// Libraries
import React, {Component} from 'react'
import classnames from 'classnames'

// Components
import SideBarTab from 'src/onboarding/components/side_bar/SideBarTab'
import SideBarButton from 'src/onboarding/components/side_bar/SideBarButton'
import {ComponentSpacer, Stack, Alignment} from 'src/clockface'

// Styles
import './SideBar.scss'

export enum SideBarTabStatus {
  Default = 'default',
  Error = 'error',
  Success = 'success',
  Pending = 'pending',
  Blank = 'blank',
}

interface Props {
  title: string
  children: JSX.Element[]
  visible: boolean
}

class SideBar extends Component<Props> {
  public static Tab = SideBarTab
  public static Button = SideBarButton

  public render() {
    const {title} = this.props

    return (
      <div className={this.containerClassName}>
        <div className="side-bar--container">
          <h3 className="side-bar--title">{title}</h3>
          <div className="side-bar--contents">
            <div className="side-bar--tabs">{this.childTabs}</div>
            <div className="side-bar--buttons">
              <ComponentSpacer
                stackChildren={Stack.Rows}
                align={Alignment.Left}
              >
                {this.childButtons}
              </ComponentSpacer>
            </div>
          </div>
        </div>
      </div>
    )
  }

  private get containerClassName(): string {
    const {visible} = this.props

    return classnames('side-bar', {show: visible})
  }

  private get childTabs(): JSX.Element[] {
    const {children} = this.props
    return React.Children.map(children, (child: JSX.Element) => {
      if (child.type === SideBarTab) {
        return child
      }
    })
  }

  private get childButtons(): JSX.Element[] {
    const {children} = this.props
    return React.Children.map(children, (child: JSX.Element) => {
      if (child.type === SideBarButton) {
        return child
      }
    })
  }
}

export default SideBar
