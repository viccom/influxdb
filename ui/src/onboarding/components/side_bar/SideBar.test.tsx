// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import SideBar from 'src/onboarding/components/side_bar/SideBar'

// Types
import {SideBarTabStatus as TabStatus} from 'src/onboarding/components/side_bar/SideBar'

// Constants
import {ComponentColor} from 'src/clockface'
import {IconFont} from 'src/clockface'

const onClick = jest.fn(() => {})

const childrenArray = [
  <SideBar.Tab
    label={'a'}
    key={'a'}
    id={'a'}
    active={true}
    status={TabStatus.Default}
    onClick={onClick}
  />,
  <SideBar.Tab
    label={'b'}
    key={'b'}
    id={'b'}
    active={false}
    status={TabStatus.Default}
    onClick={onClick}
  />,
  <SideBar.Button
    key="a"
    text="a"
    titleText="a"
    color={ComponentColor.Secondary}
    icon={IconFont.Download}
  />,
  <SideBar.Button
    text="b"
    key="b"
    titleText="b"
    color={ComponentColor.Default}
    icon={IconFont.Plus}
  />,
]

const setup = (override?, childrenArray = []) => {
  const props = {
    title: 'titleString',
    visible: true,
    ...override,
  }

  const wrapper = shallow(<SideBar {...props}>{childrenArray} </SideBar>)

  return {wrapper}
}

describe('SideBar', () => {
  describe('rendering', () => {
    it('renders with no children', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    }),
      it('renders with children, and renders its children', () => {
        const {wrapper} = setup(null, childrenArray)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.contains(childrenArray[0])).toBe(true)
        expect(wrapper.find(SideBar.Tab)).toHaveLength(2)
        expect(wrapper.find(SideBar.Button)).toHaveLength(2)
        expect(wrapper).toMatchSnapshot()
      })
  })
})
