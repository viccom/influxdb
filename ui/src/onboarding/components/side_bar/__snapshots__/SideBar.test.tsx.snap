// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar rendering renders with children, and renders its children 1`] = `
<div
  className="side-bar show"
>
  <div
    className="side-bar--container"
  >
    <h3
      className="side-bar--title"
    >
      titleString
    </h3>
    <div
      className="side-bar--contents"
    >
      <div
        className="side-bar--tabs"
      >
        <SideBarTab
          active={true}
          id="a"
          key=".0:$a"
          label="a"
          onClick={[MockFunction]}
          status="default"
        />
        <SideBarTab
          active={false}
          id="b"
          key=".0:$b"
          label="b"
          onClick={[MockFunction]}
          status="default"
        />
      </div>
      <div
        className="side-bar--buttons"
      >
        <ComponentSpacer
          align="left"
          stackChildren="rows"
        >
          <SideBarButton
            color="secondary"
            icon="download"
            key=".0:$a"
            text="a"
            titleText="a"
          />
          <SideBarButton
            color="default"
            icon="plus"
            key=".0:$b"
            text="b"
            titleText="b"
          />
        </ComponentSpacer>
      </div>
    </div>
  </div>
</div>
`;

exports[`SideBar rendering renders with no children 1`] = `
<div
  className="side-bar show"
>
  <div
    className="side-bar--container"
  >
    <h3
      className="side-bar--title"
    >
      titleString
    </h3>
    <div
      className="side-bar--contents"
    >
      <div
        className="side-bar--tabs"
      />
      <div
        className="side-bar--buttons"
      >
        <ComponentSpacer
          align="left"
          stackChildren="rows"
        />
      </div>
    </div>
  </div>
</div>
`;
